src/utils/daemon-utils.ts

```diff
@@ -490,14 +487,47 @@ export function waitForDaemonProcessExit(logger: Logger, loggerConfigService: Lo
                     logger.error('Error parsing zli config file. Please try logging in again with \'zli login\' to resolve this issue');
                     break;
                 }
+                case DAEMON_EXIT_CODES.USER_NOT_FOUND:
+                case DAEMON_EXIT_CODES.POLICY_EDITED_DISCONNECT:
+                case DAEMON_EXIT_CODES.POLICY_DELETED_DISCONNECT: {
+                    // don't report an error in this case -- handled by upstream processes
+                    break;
+                }
                 default: {
                     logger.error(`daemon process closed with nonzero exit code ${exitCode} -- for more details, see ${loggerConfigService.daemonLogPath()}`);
                     break;
                 }
                 }
             }
-
             resolve(exitCode);
         });
     });
+}
+
+/**
+ * Takes a daemon exit code and connection metadata and returns an appropriate error log statement. Can be used
+ * by connection handlers that receive a nonzero code.
+ *
+ * NOTE: in almost every case, if a code is handled here, it should be caught and ignored in by waitForDaemonProcessExit.
+ * Otherwise we would log two error statements.
+ * @param exitCode should be a nonzero number
+ * @param conn A `CreateUniversalConnectionResponse` that we can use to inform the user about the connection
```

Comment:
btw there is this cool thing you can use so in case `CreateUniversalConnectionResponse` gets renamed this gets renamed as well
```suggestion
 * @param conn A {@link CreateUniversalConnectionResponse} that we can use to inform the user about the connection
```